File: javascript/core/src/loop.js
// This code demonstrates a simple for loop in JavaScript
for (let i = 0; i < 10; i++) {
  console.log("Iteration number: " + i);
}

let i = 0
while (i < 5) {
  console.log("This will run until i is less than 5: " + i);
  i++;
}

do {
  console.log("This will run at least once, even if the condition is false.");
} while (false); // This will not run again because the condition is false

// This code demonstrates the use of for...of loop to iterate over an array in JavaScript.
// The for...of loop is used to iterate over iterable objects like arrays, strings, and more.
const numbers = [1, 2, 3, 4, 5];
for (const number of numbers) {
  console.log("Current number: " + number);
}

File: perl/core/bin/loop.pl
#!/usr/bin/perl
use strict;
use warnings;

# This script demonstrates the use of loops in Perl
# Loops allow you to execute a block of code multiple times
# Perl provides several types of loops: for, foreach, while, and until

# Using a for loop
print "=== For Loop ===\n";
for (my $i = 0; $i < 5; $i++) {
    print "Iteration $i\n";  # Print the current iteration number
}

# Using a foreach loop
print "\n=== Foreach Loop ===\n";
my @fruits = ("apple", "banana", "cherry");
foreach my $fruit (@fruits) {
    print "Fruit: $fruit\n";  # Print each fruit in the array
}

# Using a while loop
print "\n=== While Loop ===\n";
my $count = 0;
while ($count < 3) {
    print "Count is $count\n";  # Print the current count
    $count++;  # Increment the count
}

# Using an until loop
# opposite of while, it continues until the condition is true
print "\n=== Until Loop ===\n";
$count = 0;
until ($count >= 3) {
    print "Count is $count\n";  # Print the current count
    $count++;  # Increment the count
}

# do-while loop
print "\n=== Do-While Loop ===\n";
$count = 0;
do {
    print "Count is $count\n";  # Print the current count
    $count++;  # Increment the count
} while ($count < 2);  # Continue until count is less than 2
